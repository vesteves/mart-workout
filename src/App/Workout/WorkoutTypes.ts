export enum EImpactTag {
  LOW = "low",
  HIGH = "high",
}

export enum ELevelTag {
  ADVANCED = "advanced",
  INTERMEDIATE = "intermediate",
  BEGINNER = "beginner",
}

export interface IWorkout {
  description: string;
  duration: number;
  id: string;
  impactTag: EImpactTag;
  levelTag: ELevelTag;
  media: string;
  thumbnail: string;
  title: string;
  trainerId: string;
}
