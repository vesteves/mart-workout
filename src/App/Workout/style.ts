import styled from "styled-components";
import { EImpactTag, ELevelTag } from "./WorkoutTypes";
import { Container as ContainerApp } from "../style";

export interface IImpactTag {
  colorChange?: EImpactTag;
}

export interface ILevelTag {
  colorChange?: ELevelTag;
}

export const switchLevelTag = (levelTag: ELevelTag | null): string => {
  switch (levelTag) {
    case ELevelTag.BEGINNER:
      return "green";

    case ELevelTag.INTERMEDIATE:
      return "yellow";

    case ELevelTag.ADVANCED:
      return "red";

    default:
      return "white";
  }
};

export const Wrapper = styled.div`
  text-align: center;
  padding: 40px 0;
  background: #333;
`;

export const Container = styled(ContainerApp)``;

export const Image = styled.img`
  max-width: 100%;
`;

export const Title = styled.h1`
  color: #fff;
`;

export const Description = styled.h3`
  color: #fff;
`;

export const Duration = styled.p`
  color: #fff;
`;

export const Impact = styled.p`
  color: #fff;
`;

export const ImpactTag = styled.span<IImpactTag>`
  color: ${(props) =>
    props.colorChange === EImpactTag.HIGH ? "red" : "green"};
  font-weight: ${(props) =>
    props.colorChange === EImpactTag.HIGH ? "bold" : "inherit"};
`;

export const Level = styled.p`
  color: #fff;
`;

export const LevelTag = styled.span<ILevelTag>`
  color: ${(props) => switchLevelTag(props.colorChange || null)};
  font-weight: ${(props) =>
    props.colorChange === ELevelTag.ADVANCED ? "bold" : "inherit"};
`;

export const Video = styled.video`
  max-width: 80vw;
  height: auto;
`;
