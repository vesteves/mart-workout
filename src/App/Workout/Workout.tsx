import axios from "axios";
import { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { IWorkout } from "./WorkoutTypes";
import * as S from "./style";

export const Workout = () => {
  const { id } = useParams();
  const [workout, setWorkout] = useState<IWorkout>();
  useEffect(() => {
    axios.get("/api/workouts.json").then((res) => {
      setWorkout(res.data.find((item: IWorkout) => item.id === id));
    });
  }, [id]);
  return (
    <S.Wrapper>
      <S.Container>
        <S.Image src={workout?.thumbnail} alt="" />
        <S.Title>{workout?.title}</S.Title>
        <S.Description>{workout?.description}</S.Description>
        <S.Duration>Duration: {workout?.duration}</S.Duration>
        <S.Impact>
          Impact:{" "}
          <S.ImpactTag colorChange={workout?.impactTag}>
            {workout?.impactTag}
          </S.ImpactTag>
        </S.Impact>
        <S.Level>
          Level:{" "}
          <S.LevelTag colorChange={workout?.levelTag}>
            {workout?.levelTag}
          </S.LevelTag>
        </S.Level>
        {workout && workout.media && (
          <S.Video controls>
            <source src={workout?.media}></source>
          </S.Video>
        )}
      </S.Container>
    </S.Wrapper>
  );
};

export default Workout;
