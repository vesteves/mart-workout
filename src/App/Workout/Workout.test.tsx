import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Workout from "./Workout";

describe("<Workout /> test suite", () => {
  test("should match to snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <Workout />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
