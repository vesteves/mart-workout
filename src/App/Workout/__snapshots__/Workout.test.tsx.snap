// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Workout /> test suite should match to snapshot 1`] = `
<div>
  <div
    class="sc-hKMtZM iOJXuY"
  >
    <div
      class="sc-gsnTZi sc-eCYdqJ kPWacB"
    >
      <img
        alt=""
        class="sc-jSMfEi fFfgWV"
      />
      <h1
        class="sc-gKXOVf dGTZIr"
      />
      <h3
        class="sc-iBkjds cnjgHR"
      />
      <p
        class="sc-ftvSup igKBGa"
      >
        Duration: 
      </p>
      <p
        class="sc-papXJ eTZEWr"
      >
        Impact:
         
        <span
          class="sc-jqUVSM fJpjMg"
        />
      </p>
      <p
        class="sc-kDDrLX cNjBYx"
      >
        Level:
         
        <span
          class="sc-iqcoie fRcVmS"
        />
      </p>
    </div>
  </div>
</div>
`;
